### PR TITLE
Additional cleanups.

### PR DESCRIPTION
**Pull request by nashef**
This includes a variety of miscellaneous code cleanups.

- Adds vim modelines to most files.
- Removes local min/max functions in favor of `std::min` and `std::max`
- Removes "portability" libraries for systems we don't care about and probably don't work anymore.
- Remove unused functions, declarations, and use `<cassert>` instead of hand-rolled.
- Remove various unused and silly `#define`s, including `EMPTY` and `NEW` (eek).

### CODE REVIEW

File: rosette/src/Reader.cc [hunk 1]
```diff
@@ -1352,20 +1352,21 @@ Ob* Reader::resumeExpr() {
         int c = fgetc(my->file);
 #endif
 
-        if (c == EOF && RBL_WOULDBLOCK)
+        if (EOF == c && EWOULDBLOCK == errno) {
```
Comment (kjmcqueen) [inline rosette/src/Reader.cc]: Can we have some parens here?

File: rosette/src/Interrupt.cc [hunk 2]
```diff
@@ -102,7 +102,6 @@ void RosetteIoHandler(VM_EVENT type, int fd, void*) {
 #endif
 
     if (type == VM_READ_EVENT) {
-        DO_UNGETC;
```
Comment (kjmcqueen) [inline rosette/src/Interrupt.cc]: I'm not sure what this did, but if the system works without it, I approve.

File: rosette/h/BinaryOb.h [hunk 3]
```diff
@@ -72,15 +72,15 @@ class ByteVec : public BinaryOb {
         return p[n];
     }
 
-    int numberOfBytes(EMPTY) { return byteCount; }
```
Comment (kjmcqueen) [inline rosette/h/BinaryOb.h]: I think the official way of declaring this is to use the "void" keyword.  But I'm okay with it either way.

File: rosette/src/Interrupt.cc (hunk 2 above)
Comment (nashef) [reply rosette/src/Interrupt.cc]: All it did was call `getc` and then `ungetc`, which used to be a common hack to deal with some bugs in various Unix I/O systems, but is definitely unnecessary on modern Linux.

File: rosette/h/BinaryOb.h (hunk 3 above)
Comment (nashef) [reply rosette/h/BinaryOb.h]: http://david.tribble.com/text/cdiffs.htm#C99-empty-parm

File: rosette/src/Reader.cc (hunk 1 above)
Comment (nashef) [reply rosette/src/Reader.cc]: Parents aren't really necessary if you put constants on the left. Do you feel they would improve readability? I'm ambivalent, but generally prefer to use the unambiguous syntax that includes the fewest tokens.

File: rosette/h/BinaryOb.h (hunk 3 above)
Comment (kjmcqueen) [reply rosette/h/BinaryOb.h]: Since we're c++ only it doesn't matter; a matter of preference.

File: rosette/src/Interrupt.cc (hunk 2 above)
Comment (kjmcqueen) [reply rosette/src/Interrupt.cc]: Ahh.  Good to get rid of it then.  I've just never seen this particular idiom before.

File: rosette/src/Reader.cc (hunk 1 above)
Comment (kjmcqueen) [reply rosette/src/Reader.cc]: I got tired of looking up operator precedence years ago, and just prefer to see parentheses. I don't feel strongly enough about it to make you change it though.

**Pull request merged**